src/Symfony/Component/HttpKernel/ControllerMetadata/ArgumentMetadataFactoryInterface.php

```diff
@@ -0,0 +1,27 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\HttpKernel\ControllerMetadata;
+
+/**
+ * Builds method argument data.
+ *
+ * @author Iltar van der Berg <kjarli@gmail.com>
+ */
+interface ArgumentMetadataFactoryInterface
+{
+    /**
+     * @param mixed $controller The controller to resolve the arguments for.
```

Comment:
I think it's for, because it doesn't resolve it for the controller but based on. (I'm not native either so correct me if I'm wrong)